python/example_load_gen.py

```diff
@@ -0,0 +1,307 @@
+
+        get_start_time = perf_counter_ns()
+        get_result: Optional[CacheGetResponse] = await self.execute_request_and_update_context_counts(
+            context,
+            lambda: client.get(self.cache_name, cache_key)
+        )
+        if get_result:
+            get_duration = self.get_elapsed_millis(get_start_time)
+            context.get_latencies.record_value(get_duration)
+            if get_result.status() == CacheGetStatus.HIT:
+                value = get_result.value()
+                value_string = f"{value[0:10]}... (len: {len(value)})"
+            else:
+                value_string = 'n/a'
+
+            if context.global_request_count % 1_000 == 0:
+                self.logger.info(
+                    f"worker: {worker_id}, worker request: {operation_id}, global request: {context.global_request_count}, status: {get_result.status()}, val: {value_string}"
+                )
+
+    T = TypeVar("T")
+
+    async def execute_request_and_update_context_counts(
+            self,
+            context: BasicPythonLoadGenContext,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Optional[T]:
+        result, response = await self.execute_request(block)
+        self.update_context_counts_for_request(context, result)
+        return response
+
+    async def execute_request(
+            self,
+            block: Callable[[], Coroutine[None, None, T]]
+    ) -> Tuple[AsyncSetGetResult, Optional[T]]:
+        try:
+            result = await block()
+            return AsyncSetGetResult.SUCCESS, result
+        except InternalServerError as e:
+            # TODO verify exception type
+            self.logger.error(f"Caught InternalServerError: {e}")
+            return AsyncSetGetResult.UNAVAILABLE, None
+        except TimeoutError as e:
+            # TODO need to verify exception type
+            self.logger.error(f"Caught TimeoutError: {e}")
+            return AsyncSetGetResult.DEADLINE_EXCEEDED, None
+
+    @staticmethod
+    def update_context_counts_for_request(
+            context: BasicPythonLoadGenContext,
+            result: AsyncSetGetResult
+    ):
+        context.global_request_count += 1
+        if result == AsyncSetGetResult.SUCCESS:
+            context.global_success_count += 1
+        elif result == AsyncSetGetResult.UNAVAILABLE:
+            context.global_unavailable_count += 1
+        elif result == AsyncSetGetResult.DEADLINE_EXCEEDED:
+            context.global_deadline_exceeded_count += 1
+        else:
+            raise ValueError(f"Unsupported result type: {result}")
+
+    def tps(self, context: BasicPythonLoadGenContext, request_count: int) -> int:
+        return round((request_count * 1000) / self.get_elapsed_millis(context.start_time))
+
+    @staticmethod
+    def percent_requests(context: BasicPythonLoadGenContext, count: int) -> float:
+        return round((count / context.global_request_count) * 100 * 10) / 10
```

Comment:
what are the 10's for here? Shouldn't this just be:
```javascript
round((count / context.global_request_count) * 100)
```

Reply:
one of the divisions is outside of the `round`, in order to retain the first digit after the decimal point.  The other two could be combined but I was trying to make it clear that it was `*100` to convert to percentage, and then `* 10` to allow us to preserve the first decimal point.  Can add a comment.

Reply:
left a comment in 21d518923c08238302de0e298e415624bcd2273b

Reply:
wouldn't it be better to print the actual real number using `f'{tps(context, count):.01f}'` if that's your intent?